ui/src/tool-panel/location-track/splitting/location-track-splitting-infobox.tsx

```diff
@@ -0,0 +1,290 @@
+        errors.push({
+            field: 'name',
+            reason: 'conflicts-with-track',
+            type: ValidationErrorType.ERROR,
+        });
+    }
+    return errors;
+};
+
+const validateSplitDescription = (
+    description: string,
+    duplicateOf: LocationTrackId | undefined,
+) => {
+    const errors: ValidationError<Split>[] = validateLocationTrackDescriptionBase(description);
+    if (!duplicateOf && description === '')
+        errors.push({
+            field: 'descriptionBase',
+            reason: 'mandatory-field',
+            type: ValidationErrorType.ERROR,
+        });
+    return errors;
+};
+
+export const LocationTrackSplittingInfoboxContainer: React.FC<
+    LocationTrackSplittingInfoboxContainerProps
+> = ({
+    locationTrackId,
+    initialSplit,
+    splits,
+    duplicateLocationTracks,
+    visibilities,
+    visibilityChange,
+    allowedSwitches,
+    removeSplit,
+    cancelSplitting,
+    updateSplit,
+    showArea,
+}) => {
+    const locationTrack = useLocationTrack(locationTrackId, 'DRAFT');
+    const [startAndEnd, _] = useLocationTrackStartAndEnd(locationTrackId, 'DRAFT');
+    const conflictingTracks = useConflictingTracks(
+        locationTrack?.trackNumberId,
+        [initialSplit, ...splits].map((s) => s.name),
+        [initialSplit, ...splits].map((s) => s.duplicateOf).filter(filterNotEmpty),
+        'DRAFT',
+    );
+
+    return (
+        locationTrack &&
+        startAndEnd && (
+            <LocationTrackSplittingInfobox
+                duplicateLocationTracks={duplicateLocationTracks}
+                visibilities={visibilities}
+                visibilityChange={visibilityChange}
+                initialSplit={initialSplit}
+                splits={splits}
+                allowedSwitches={allowedSwitches}
+                removeSplit={removeSplit}
+                cancelSplitting={cancelSplitting}
+                updateSplit={updateSplit}
+                startAndEnd={startAndEnd}
+                conflictingLocationTracks={conflictingTracks?.map((t) => t.name) || []}
+                locationTrack={locationTrack}
+                showArea={showArea}
+            />
+        )
+    );
+};
+
+export const LocationTrackSplittingInfobox: React.FC<LocationTrackSplittingInfoboxProps> = ({
+    duplicateLocationTracks,
+    visibilities,
+    visibilityChange,
+    initialSplit,
+    splits,
+    allowedSwitches,
+    removeSplit,
+    cancelSplitting,
+    updateSplit,
+    startAndEnd,
+    conflictingLocationTracks,
+    locationTrack,
+    showArea,
+}) => {
+    const { t } = useTranslation();
+    const getSplitAddressPoint = (split: Split): AddressPoint | undefined => {
+        const switchAtSplit = allowedSwitches.find((s) => s.switchId === split.switchId);
+
+        if (switchAtSplit?.location && switchAtSplit?.address) {
+            return {
+                point: { ...switchAtSplit.location, m: -1 },
```

Comment:
(kind of a nitpick): Sanoisin, että jos tällaisia maagisia lukuja joutuu käyttämään, niin ne kannattaisi kyllä constittaa tai tyypittää, ettei esim. käyttö-/vertailupaikkoihin päätyisi myös maagisia lukuja (heti tai tulevaisuudessa), joiden vaihtaminen/löytäminen esim. mahdollisen refaktoroinnin yhteydessä on hyvin raskasta. Samalla constitus dokumentoisi mitä erikoisarvolla tarkoitetaan.